disnake/permissions.py

```diff
@@ -131,7 +154,7 @@ def __init__(self, permissions: int = 0, **kwargs: bool):
                 raise TypeError(f"{key!r} is not a valid permission name.")
             setattr(self, key, value)
 
-    def is_subset(self, other: Permissions) -> bool:
+    def is_subset(self, other: Self) -> bool:
         """Returns ``True`` if self has the same or fewer permissions as other."""
         if isinstance(other, Permissions):
```

Comment:
if the `other` annotation is changed to `Self`, this should check for `type(self)` instead of `Permissions` - see the other comment about removing these entirely as well though

Reply:
Reverted the typing change as changing the checks would be breaking